paddle/fluid/framework/framework.proto

```diff
@@ -91,52 +91,54 @@ message OpProto {
   required string comment = 5;
 }
 
-enum DataType {
-  BOOL = 0;
-  INT16 = 1;
-  INT32 = 2;
-  INT64 = 3;
-  FP16 = 4;
-  FP32 = 5;
-  FP64 = 6;
-}
+message VarType {
+  enum Type {
+    // POD types
+    BOOL = 0;
+    INT16 = 1;
+    INT32 = 2;
+    INT64 = 3;
+    FP16 = 4;
+    FP32 = 5;
+    FP64 = 6;
+
+    // other types, that might need additonal descriptions
+    LOD_TENSOR = 7;    // LoDTensor
+    SELECTED_ROWS = 8; // Tensor
+    FEED_MINIBATCH = 9;
+    FETCH_LIST = 10;
+    STEP_SCOPES = 11;
+    LOD_RANK_TABLE = 12;
+    // LOD_TENSOR_ARRAY = 7; // replaced by array-of-tensors. c.f. ArrayType
+    // PLACE_LIST = 8; // replaced by array-of-place. c.f. ArrayType
+    READER = 13; // ReaderType
+  }
 
-message TensorDesc {
-  required DataType data_type = 1;
-  repeated int64 dims = 2; // [UNK, 640, 480] is saved as [-1, 640, 480]
-}
+  required Type type = 1;
 
-message LoDTensorDesc {
-  required TensorDesc tensor = 1;
-  optional int32 lod_level = 2 [ default = 0 ];
-}
+  message Tensor {
+    required Type data_type = 1;
+    repeated int64 dims = 2; // [UNK, 640, 480] is saved as [-1, 640, 480]
+  }
+  optional Tensor selected_rows = 2;
 
-message LoDTensorArrayDesc {
-  required TensorDesc tensor = 1;
-  optional int32 lod_level = 2 [ default = 0 ];
-}
+  message LoDTensor {
+    required Tensor tensor = 1;
+    optional int32 lod_level = 2 [ default = 0 ];
+  }
+  optional LoDTensor lod_tensor = 3;
 
-message ReaderDesc { repeated LoDTensorDesc lod_tensor = 1; }
+  message Array { VarType elem_type = 1; }
+  optional Array array = 4;
+
+  message Reader { repeated LoDTensor lod_tensor = 1; }
+  optional Reader reader = 5;
+}
 
 message VarDesc {
-  enum VarType {
-    LOD_TENSOR = 1;
-    SELECTED_ROWS = 2;
-    FEED_MINIBATCH = 3;
-    FETCH_LIST = 4;
-    STEP_SCOPES = 5;
-    LOD_RANK_TABLE = 6;
-    LOD_TENSOR_ARRAY = 7;
-    PLACE_LIST = 8;
-    READER = 9;
-  }
   required string name = 1;
   required VarType type = 2;
```

Comment:
Can `VarDesc::type` be POD type like `BOOL`, `INT16`? Or it can only be `LOD_TENSOR`, `SELECTED_ROWS`...?

Reply:
VarType::Type contains POD type.